File: Programmers/level 1/신고 결과 받기.js
function solution(id_list, report, k) {
    const usersCnt = id_list.length;
    const answer = new Array(usersCnt).fill(0);
    const reportArray = Array.from(Array(usersCnt), ()=>new Array())

    for (let i=0; i<report.length; i++) {
        const user = report[i].split(" ")[0];
        const reported = report[i].split(" ")[1];
        
        //console.log(id_list.indexOf(user));
        const idx = id_list.indexOf(reported);
        if (reportArray[idx].findIndex((e)=>e===user)==-1) {
            reportArray[idx].push(user);
        }
    }
    

    for (let i=0; i<reportArray.length; i++) {
        if (reportArray[i].length<k) continue;
        for (const user of reportArray[i]) {
            const idx = id_list.indexOf(user);
            answer[idx]++;
        }
    }
     
    return answer;
}

File: Programmers/level 1/신고 결과 받기.cpp
#include <string>
#include <vector>
#include <sstream>
#include <map>
#include <algorithm>
#include <iostream>

using namespace std;

vector<int> solution(vector<string> id_list, vector<string> report, int k) {
    int n=id_list.size();
    vector<int> answer(n, 0);
    vector<vector<string>> v(n, vector<string>(0));
    map <string,int> m;
    
    for (int i=0; i<report.size(); i++) {
        vector<string> result;
    	stringstream ss(report[i]);
    	string temp;
    	while (getline(ss, temp, ' ')) {
        	result.push_back(temp);
    	}
        string user=result[0];
        string report=temp;
        
        int idx=find(id_list.begin(), id_list.end(), user)-id_list.begin();
        
        if (find(v[idx].begin(), v[idx].end(), report)!=v[idx].end()) continue;
        
        v[idx].push_back(report);
        m[report]++;
    }
    
    for (int i=0; i<n; i++) {
        for (int j=0; j<v[i].size(); j++) {
            if (m[v[i][j]]>=k) answer[i]++;
        }
    }
    
    return answer;
}
